src/clr-angular/utils/drag-and-drop/providers/drag-event-listener.spec.ts

```diff
@@ -0,0 +1,344 @@
+
+const expectEventPropValues = <T>(event: ClrDragEvent<T>) => {
+    return {
+        toBe: (element: Node, pageX: number, pageY: number, dragTransfer?: T, dropGroup?: string|string[]) => {
+            expect(event.draggableElement).toBe(element);
+            expect(event.dragPosition.pageX).toBe(pageX);
+            expect(event.dragPosition.pageY).toBe(pageY);
+
+            if (dropGroup) {
+                expect(event.dragScope).toEqual(dropGroup);
+            } else {
+                expect(event.dragScope).toBeUndefined();
+            }
+
+            if (dragTransfer) {
+                expect(event.dragDataTransfer).toEqual(dragTransfer);
+            } else {
+                expect(event.dragDataTransfer).toBeUndefined();
+            }
+        }
+    };
+};
+
+export default function(): void {
+    describe("Drag Event Listener", function() {
+        let dragEventListenerService: ClrDragEventListener<DragTransfer>;
+        let dragAndDropEventBusService: ClrDragAndDropEventBus<DragTransfer>;
+
+        let fixture: ComponentFixture<any>;
+        let testComponent: TestComponent;
+
+        let draggableButton: Node;
+
+        beforeEach(function() {
+            TestBed.configureTestingModule({declarations: [TestComponent], providers: [ClrDragAndDropEventBus]});
+
+            fixture = TestBed.createComponent(TestComponent);
+            testComponent = fixture.componentInstance;
+
+            fixture.detectChanges();
+
+            dragEventListenerService =
+                <ClrDragEventListener<DragTransfer>>fixture.debugElement.injector.get(ClrDragEventListener);
+            dragAndDropEventBusService = TestBed.get(ClrDragAndDropEventBus);
+
+            draggableButton = testComponent.draggableButtonRef.nativeElement;
+        });
+
+        afterEach(() => {
+            fixture.destroy();
+        });
+
+        function testCases(startEvent: string, moveEvent: string, endEvent: string) {
+            it(startEvent + " alone shouldn't fire any drag event", function() {
+                emulateEventOn(startEvent, 0, 0, draggableButton);
+                expect(testComponent.dragStartFired).toBeFalsy();
+                expect(testComponent.dragMoveFired).toBeFalsy();
+                expect(testComponent.dragEndFired).toBeFalsy();
+            });
+
+            it("first " + startEvent + " and first " + moveEvent + " should fire only dragstart event", function() {
+                emulateEventOn(startEvent, 0, 0, draggableButton);
+                emulateEventOn(moveEvent, 22, 33);
+
+                expect(testComponent.dragStartFired).toBeTruthy();
+                expectEventPropValues(testComponent.dragEvent).toBe(draggableButton, 22, 33, null);
+
+                expect(testComponent.dragMoveFired).toBeFalsy();
+                expect(testComponent.dragEndFired).toBeFalsy();
+            });
+
+            it("multiple " + moveEvent + " after " + startEvent + " should fire dragstart event", function() {
```

Comment:
looks like this test is redundant. the above test already tests this, no?

Reply:
this one is removed.